components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/AbstractApplicationAuthenticator.java

```diff
@@ -352,6 +360,79 @@ protected String getUserStoreAppendedName(String userName) {
         return userName;
     }
 
+    protected org.wso2.carbon.user.core.common.User getUser(AuthenticatedUser authenticatedUser)
+            throws AuthenticationFailedException {
+
+        org.wso2.carbon.user.core.common.User user = null;
+        String tenantDomain = authenticatedUser.getTenantDomain();
+        if (tenantDomain == null) {
+            return null;
+        }
+        int tenantId = IdentityTenantUtil.getTenantId(tenantDomain);
+        try {
+            UserRealm userRealm = FrameworkServiceDataHolder.getInstance().getRealmService()
+                    .getTenantUserRealm(tenantId);
+            if (userRealm != null) {
+                UserStoreManager userStoreManager = userRealm.getUserStoreManager();
+                List<org.wso2.carbon.user.core.common.User> userList
+                        = ((AbstractUserStoreManager) userStoreManager).getUserListWithID(
+                        USERNAME_CLAIM, authenticatedUser.getUserName(), null);
+                if (userList.isEmpty()) {
+                    userList = ((AbstractUserStoreManager) userStoreManager).getUserListWithID(
+                            EMAIL_ADDRESS_CLAIM, authenticatedUser.getUserName(), null);
+                }
+                userList = getValidUsers(userList);
+                if (CollectionUtils.isEmpty(userList)) {
+                    return null;
+                }
+                if (userList.size() > 1) {
+                    if (log.isDebugEnabled()) {
+                        log.debug("There are more than one user with the provided username claim value: "
```

Comment:
better to throw an exception for this rather than printing it as a debug log

Reply:
Fixed with 3299151e890857486827301005d986c6fdc7bb90